src/hostapi/pulseaudio/pa_linux_pulseaudio_cb.c

```diff
@@ -717,16 +760,10 @@ PaError PaPulseAudio_StartStreamCb( PaStream * s )
         {
             PaPulseAudio_Lock( pulseaudioHostApi->mainloop );
             /* Zero means success */
-            if( ! pa_stream_connect_record( stream->inputStream,
+            if( pa_stream_connect_record( stream->inputStream,
                                             pulseaudioName,
                                             &stream->inputBufferAttr,
                                             pulseaudioStreamFlags ) )
-            {
-                pa_stream_set_started_callback( stream->inputStream,
-                                                PaPulseAudio_StreamStartedCb,
-                                                stream );
-            }
-            else
             {
                 PA_DEBUG( ("Portaudio %s: Can't read audio!\n",
                           __FUNCTION__) );
```

Comment:
```suggestion
                          __FUNCTION__) );
                   PaPulseAudio_UnLock( pulseaudioHostApi->mainloop );
```
UnLock in error else-case is missing.